tests/test_edge_cases.py

```diff
@@ -1339,6 +1338,39 @@ class TopModel(BaseModel):
     assert m.nest.modified_number == 1
 
 
+def test_init_model_validate():
+    calls = []
+
+    class MyModel(BaseModel):
+        number: int = 1
+
+        def __init__(self, **kwargs):
+            super().__init__(**kwargs)
+            self.number += 1
+            calls.append(str(kwargs))
+
+    m = MyModel(number=1)
+    assert m.number == 2
+    assert calls == ["{'number': 1}"]
+
+    m = MyModel.model_validate(dict(number=2))
+    assert m.number == 3
+    # insert_assert(calls)
+    assert calls == [
+        "{'number': 1}",
+        # NOTE! model_dict has been modified in-place
+        "{'validated_data': ValidatedData(model_dict={'number': 3}, fields_set={'number'})}",
```

Comment:
Is there any potential for collisions if someone uses their own keyword argument with the same name? If it's checking that the type is `ValidatedData` then that's probably fine.

Reply:
There's a test for that on pydantic-core, it's fine.